test/models/secret_storage_test.rb

```diff
@@ -93,27 +147,45 @@
   end
 
   describe SecretStorage::HashicorpVault do
-    # note, we need to call the storage engine here directly
-    # as the model has already loaded it's config
-    # from ENV
+    let(:response_headers) { {'Content-Type': 'application/json'} }
+
+    describe ".client" do
+      it 'creates a valid client' do
+        assert_instance_of(VaultClient, SecretStorage::HashicorpVault.vault_client)
+      end
+    end
+
     describe ".read" do
       before do
-        data = {data: { vault:"bar"}}.to_json
-        stub_request(:get, "https://127.0.0.1:8200/v1/secret%2Ffoo%252Fisbar").
-          to_return(status: 200, body: data, headers: {'Content-Type': 'application/json'})
         fail_data = {data: { vault:nil}}.to_json
         # client gets a 200 and nil body when key is missing
-        stub_request(:get, "https://127.0.0.1:8200/v1/secret%2Fnotgoingtobethere").
+        stub_request(:get, "https://127.0.0.1:8200/v1/secret/this/key/isnot/there").
           to_return(status: 200, body: fail_data, headers: {'Content-Type': 'application/json'})
+        # this is the auth request, just needs to return 200 for our purposes
+        stub_request(:post, "https://127.0.0.1:8200/v1/auth/cert/login")
+        # using the stubbed client
+        stub_request(:get, "https://127.0.0.1:8200/v1/secret/production/foo/pod2/isbar").
+          to_return(status: 200, headers: {'Content-Type': 'application/json'})
+        stub_request(:get, "https://127.0.0.1:8200/v1/secret/notgoingtobethere").
+          to_return(status: 200, headers: {'Content-Type': 'application/json'})
+        not_branch = ['is_now_a_leaf']
+        stub_request(:get, "https://127.0.0.1:8200/v1/secret/?list=true").
+          to_return(status: 200, body: not_branch, headers: response_headers)
```

Comment:
does every read test need all of these stubs or can we put them into the 1 that needs them / make a nested describe with those that need all ?